libnavigation-core/src/test/java/com/mapbox/navigation/core/routealternatives/AlternativeRouteProgressDataProviderTest.kt

```diff
@@ -107,6 +107,32 @@ class AlternativeRouteProgressDataProviderTest {
         assertEquals(expected, actual)
     }
 
+
+    @Test
+    fun `before fork, alternative starts after the primary route ahead of current position`() =
```

Comment:
THB, I didn't get the case 🙂

Reply:
The alternative starts along the route but **ahead** of the current position. So it's kinda not started yet. We are not yet on this alternative.

Reply:
😱 
How is this possible? Does it mean that continuous alternatives work in a strange way, or it means that user calculated an alternative using some other mechanism?
Should the SDK accept such alternatives at all? I guess if user switches to this alternative, the SDK does reroute immediately, doesn't it?

Reply:
NN accepts such alternatives, they said it's valid.
Possible scenario:
I think I know what’s going on (at least for the uuid I was investigating), considering that I’ve only seen refreshes for an alternatives with ids: “uuid#1“, “uuid#2“. 
1. Request routes, receive “uuid#0“, "uuid#1", "uuid#2";
2. Make "uuid#0" primary, others - alternatives;
2. Go off_route;
3. We make a new request and receive a new “primary“ route, which just tells us to return to the position the original route started from (as the alternatives do);
4. Old alternatives stay the same: they are ahead of our current position.